src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -66,6 +66,10 @@
     @Query("select user from User user")
     Page<User> findAllWithGroups(Pageable pageable);
 
+    @EntityGraph(attributePaths = { "groups" })
+    @Query("select user from User user where user.login like %:partialLogin% or user.email like %:partialEmail%")
+    Page<User> searchByLoginOrNameWithGroup(String partialLogin, String partialEmail, Pageable pageable);
```

Comment:
```suggestion
    Page<User> searchByLoginOrNameWithGroups(String partialLogin, String partialEmail, Pageable pageable);
```

Reply:
Ich baue das selber in meinen nächsten Commit ein.